src/commonMain/kotlin/fr/acinq/lightning/crypto/Bolt3Derivation.kt

```diff
@@ -4,41 +4,45 @@ import fr.acinq.bitcoin.ByteVector32
 import fr.acinq.bitcoin.Crypto.sha256
 import fr.acinq.bitcoin.PrivateKey
 import fr.acinq.bitcoin.PublicKey
-import fr.acinq.lightning.utils.leftPaddedCopyOf
 
-object Generators {
-
-    fun fixSize(data: ByteArray): ByteVector32 = when {
-        data.size == 32 -> ByteVector32(data)
-        data.size < 32 -> ByteVector32(data.leftPaddedCopyOf(32))
-        else -> error("Data is too big")
-    }
+/**
+ * BOLT 3 Key derivation scheme.
+ */
+object Bolt3Derivation {
 
     fun perCommitSecret(seed: ByteVector32, index: Long): PrivateKey = PrivateKey(ShaChain.shaChainFromSeed(seed, 0xFFFFFFFFFFFFL - index))
 
     fun perCommitPoint(seed: ByteVector32, index: Long): PublicKey = perCommitSecret(seed, index).publicKey()
 
-    fun derivePrivKey(secret: PrivateKey, perCommitPoint: PublicKey): PrivateKey {
+    private fun derivePrivKey(secret: PrivateKey, perCommitPoint: PublicKey): PrivateKey {
         // secretkey = basepoint-secret + SHA256(per-commitment-point || basepoint)
         return secret + (PrivateKey(sha256(perCommitPoint.value + secret.publicKey().value)))
     }
 
-    fun derivePubKey(basePoint: PublicKey, perCommitPoint: PublicKey): PublicKey {
+    fun PrivateKey.derive(perCommitPoint: PublicKey): PrivateKey = derivePrivKey(this, perCommitPoint)
```

Comment:
I'm not sure we should define this directly on `PrivateKey`, it means that when we use this we don't necessarily know what kind of derivation this is doing (this is a very custom type of key derivation). I'd rather explicitly call `Bolt3Derivation.derivePrivKey` to keep this explicit. Same thing for `PublicKey.derive`.

Or maybe we could keep these extension methods but name them more explicitly? For example `PrivateKey.deriveForCommitment`?

Reply:
Went with `deriveForCommitment`/`deriveForRevocation` in cc3eaac9.